### PR TITLE
PositionControl: skip controller only if adjusted thrust setpoint are…

### PR DESCRIPTION
**Pull request by Stifael**
* consider landed if not armed
* vehicle_local_position_setpiont publication: follow the logic [here](https://github.com/PX4/Firmware/pull/11056/files#diff-fc77c3ef569029d45764664c75d4b0c1R776) to improve logging analysis
* do not publish the adjusted thrust setpoint from ground-contact/maybe-landed to simplify landed/maybe-landed analysis
* initialize the landed-structure with safe values

### CODE REVIEW

**Comment by MaEtUgR** [comment]
rebased for the fmu-v2 savings and hopefully CI passing

**Comment by Stifael** [comment]
@MaEtUgR, not related.

@bkueng, with this change the vehicle_local_position_setpoint.x/y will be NAN during manual-position control and fast forward flight. As discussed, this would require to adjust Flightreview, otherwise there will be a bunch of NANs

File: src/modules/mc_pos_control/PositionControl.cpp [hunk 1]
```diff
@@ -55,8 +55,25 @@ void PositionControl::updateState(const PositionControlStates &states)
 	_vel_dot = states.acceleration;
 }
 
+void PositionControl::_setCtrlFlagTrue()
```
Comment (bkueng) [inline src/modules/mc_pos_control/PositionControl.cpp]: How about `void PositionControl::_setCtrlFlag(bool value)`?

Comment (Stifael) [reply src/modules/mc_pos_control/PositionControl.cpp]: true. not sure what I was thinking

Comment (Stifael) [reply src/modules/mc_pos_control/PositionControl.cpp]: @MaEtUgR will do the change. thanks

**Comment by MaEtUgR** [comment]
@bkueng's review comment will be addressed in my follow up pr.

**Comment by bramsvs** [comment]
>     * do not publish the adjusted thrust setpoint from ground-contact/maybe-landed to simplify landed/maybe-landed analysis

@Stifael Quick post-merge question... How exactly does this PR simplify landed/maybe-landed analysis?

I'm asking because I just had a case where ground-contact was detected inadvertently, causing the xy thrust set to zero resulting in a crash. If the thrust is logged unchanged this will be even harder to debug..(?)

**Comment by Stifael** [comment]
@bramsvs with this PR vehicle_local_position_setpoint now always agrees with the logic [here](https://github.com/PX4/Firmware/pull/11056/files#diff-fc77c3ef569029d45764664c75d4b0c1R776), even during landing. However, that does not change anything in terms of landing behavior. It is to note, that the vehicle_local_position_setpoint topic is so far only used to detect vehicle intention (upwards or downards motion). The vehicle_local_position_setpoint.thrust[0..2] is only used for logging. This means that when you look at vehicle_local_position_setpoint topic, the different messages should now agree with what the controller is using and spits out. 
The actual thrust used in the  system and sent to the attitude-controller is vehicle_attitude_setpoint.thrust. The actuator_control.thrust msg (thats the message before entering the mixer), is used in the land-detector. Hence, if you want to see if the thrust is actually at zero during ground-contact, the you should look at actuator_control topic

**Comment by julianoes** [comment]
Found a regression: #11197

**Pull request merged**